Caculator_G/Caculator_G/ViewController.swift

```diff
@@ -11,21 +11,85 @@ import UIKit
 class ViewController: UIViewController {
     
     @IBOutlet weak var numLabel: UILabel!
+    var labelFontSize : CGFloat = 70
+    var numValueText : String = "0"
+    var numDisplayText : String = "0"
+    var isNegative : Bool = false
     
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
     }
 
     @IBAction func touchNumber(_ sender: UIButton) {
-        let numText = sender.titleLabel?.text
-        if numLabel.text == "0"{
-            numLabel.text = numText
+        let numBtnText = sender.titleLabel?.text
+        if numLabel.text == "0" || numLabel.text == "-0"{
+            numValueText = numBtnText!
+            numDisplayText = numBtnText!
+            if isNegative == true{
+                numDisplayText.insert("-", at: numDisplayText.startIndex)
+            }
+            numLabel.text = numDisplayText
         }
-        else{
-            numLabel.text! += numText!
+        else if numValueText.count <= 8 {
+            numValueText += numBtnText!
+            numDisplayText = numValueText
+            addCommaText(count: numValueText.count, numDisplayText: &numDisplayText)
+            resizeLabelFont(count: numValueText.count, labelFontSize: &labelFontSize)
+            if isNegative == true{
+                numDisplayText.insert("-", at: numDisplayText.startIndex)
+            }
+            numLabel.text! = numDisplayText
+            numLabel.font = UIFont.systemFont(ofSize: labelFontSize)
         }
     }
     
+    @IBAction func resetLabel(_ sender: UIButton) {
+        numLabel.text = "0"
+        numDisplayText = "0"
+        numValueText = "0"
+        isNegative = false
+        labelFontSize = 70
+        numLabel.font = UIFont.systemFont(ofSize: labelFontSize)
+    }
+    
+    
+    @IBAction func changeSign(_ sender: UIButton) {
+        isNegative.toggle()
+        if isNegative == true{
+            numDisplayText.insert("-", at: numDisplayText.startIndex)
+        }
+        else
+        {
+            numDisplayText.remove(at: numDisplayText.startIndex)
+        }
+        numLabel.text! = numDisplayText
+    }
 }
 
+func addCommaText(count:Int, numDisplayText : inout String)
```

Comment:
```suggestion
func addCommaText(count: Int, numDisplayText: inout String)
```

Swift Style guide에서는 콜론 뒤에서만 띄어쓰기가 지향되어요 :)